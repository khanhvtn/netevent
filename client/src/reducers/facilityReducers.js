import {
    FACILITY_LOADING,
    CREATE_FACILITY,
    FETCH_ALL_FACILITIES,
    UPDATE_FACILITY,
    DELETE_FACILITY,
    FACILITY_CREATE_SUCCESS,
    FACILITY_UPDATE_SUCCESS,
} from '../constants';

const initialState = {
    isLoading: false,
    isCreated: false,
    isUpdated: false,
    facility: null,
    facilities: [],
};

export default function errorReducers(state = initialState, action) {
    switch (action.type) {
        case FACILITY_LOADING:
            return { ...state, isLoading: action.payload };
        case CREATE_FACILITY:
            return { ...state, facility: action.payload };
        case FACILITY_CREATE_SUCCESS:
            return { ...state, isCreated: action.payload };
        case FACILITY_UPDATE_SUCCESS:
            return { ...state, isUpdated: action.payload };
        case FETCH_ALL_FACILITIES:
            return { ...state, facilities: action.payload };
        case UPDATE_FACILITY:
            return {
                ...state,
                facilities: state.facilities.map(
                    (facility) => action.payload._id === facility._id ? action.payload : facility)
            };
        case DELETE_FACILITY:
            return {
                ...state,
                facilities: state.facilities.filter(
                    (facility) => action.payload !== facility._id
                )
            }
        default:
            return state;
    }
}
