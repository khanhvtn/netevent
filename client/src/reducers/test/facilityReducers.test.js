import {
    FACILITY_GET_ALL,
    FACILITY_LOADING,
    FACILITY_CREATE_SUCCESS,
    FACILITY_UPDATE_SUCCESS,
    FACILITY_DELETE_SUCCESS,
    FACILITY_GET_ALL_FILTER,
} from '../../constants';
import facilityReducers from '../facilityReducers'
import expect from 'expect'

describe('FACILITY REDUCER', () => {
    const initialState = {
        isLoading: false,
        facilities: [],
        totalPages: null,
        createSuccess: false,
        updateSuccess: false,
        deleteSuccess: false,
    };
    it('should return default state', () => {
        const action = facilityReducers(undefined, {})
        expect(action).toMatchSnapshot(initialState)
    });
    it('should return FACILITY_LOADING', () => {
        const action = facilityReducers(initialState, {
            type: FACILITY_LOADING
        })
        expect(action).toMatchSnapshot({
            isLoading: action.payload,
            facilities: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: false,
            deleteSuccess: false,
        })
    });
    it('should return FACILITY_CREATE_SUCCESS', () => {
        const action = facilityReducers(initialState, {
            type: FACILITY_CREATE_SUCCESS
        })
        expect(action).toMatchSnapshot({
            isLoading: false,
            facilities: [],
            totalPages: null,
            createSuccess: action.payload,
            updateSuccess: false,
            deleteSuccess: false,
        })
    });
    it('should return FACILITY_UPDATE_SUCCESS', () => {
        const action = facilityReducers(initialState, {
            type: FACILITY_UPDATE_SUCCESS
        })
        expect(action).toMatchSnapshot({
            isLoading: false,
            facilities: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: action.payload,
            deleteSuccess: false,
        })
    });
    it('should return FACILITY_DELETE_SUCCESS', () => {
        const action = facilityReducers(initialState, {
            type: FACILITY_DELETE_SUCCESS
        })
        expect(action).toMatchSnapshot({
            isLoading: false,
            facilities: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: false,
            deleteSuccess: action.payload,
        })
    });
})