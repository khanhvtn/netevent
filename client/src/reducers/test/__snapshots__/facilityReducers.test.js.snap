// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FACILITY REDUCER should return FACILITY_CREATE_SUCCESS 1`] = `
Object {
  "createSuccess": undefined,
  "deleteSuccess": false,
  "facilities": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`FACILITY REDUCER should return FACILITY_DELETE_SUCCESS 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": undefined,
  "facilities": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`FACILITY REDUCER should return FACILITY_LOADING 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "facilities": Array [],
  "isLoading": undefined,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`FACILITY REDUCER should return FACILITY_UPDATE_SUCCESS 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "facilities": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": undefined,
}
`;

exports[`FACILITY REDUCER should return default state 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "facilities": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;
